### PR TITLE
[DOXIASITETOOLS-329] Expose last modification date for Doxia source

### PR DESCRIPTION
**Pull request by kwin**
files in Velocity context

### CODE REVIEW

**Review by michael-o — CHANGES_REQUESTED** [review]
You need to update documentation as well.

File: doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java [hunk 1]
```diff
@@ -498,6 +503,16 @@ protected Context createDocumentVelocityContext(
             context.put("doxiaSiteRendererVersion", DOXIA_SITE_RENDERER_VERSION);
         }
 
+        String doxiaSourcePath = docRenderingContext.getDoxiaSourcePath();
+        if (doxiaSourcePath != null) {
+            Path path = Paths.get(doxiaSourcePath);
+            try {
+                FileTime lastModifiedTime = Files.getLastModifiedTime(path);
+                context.put("lastModificationDate", new Date(lastModifiedTime.toMillis()));
```
Comment (michael-o) [inline doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/DefaultSiteRenderer.java:511]: This should be the canonical term: "lastModifiedDate"

**Comment by michael-o** [comment]
@kwin, do you want to pick this up for the next release?

**Comment by kwin** [comment]
This approach does not work well as filesystem modification dates typically don't reflect the last modification in the SCM either and differ for the same file on different clones/working copies. Will come up with another PR leveraging the SCM last modified date.

**Comment by michael-o** [comment]
> This approach does not work well as filesystem modification dates typically don't reflect the last modification in the SCM either and differ for the same file on different clones/working copies. Will come up with another PR leveraging the SCM last modified date.

Don't forget that you can also build from a source tarball as well...

**Pull request closed without merging**